Zend/zend_types.h

```diff
@@ -428,7 +435,7 @@ struct _zend_array {
 #define HT_SIZE_TO_MASK(nTableSize) \
 	((uint32_t)(-((nTableSize) + (nTableSize))))
 #define HT_HASH_SIZE(nTableMask) \
-	(((size_t)(uint32_t)-(int32_t)(nTableMask)) * sizeof(uint32_t))
+	(((size_t)-(uint32_t)(nTableMask)) * sizeof(uint32_t))
```

Comment:
`(int32_t)(nTableMask)` is implementation defined when `nTableMask` has its max value `(uint32_t)INT32_MAX+1`.

I _think_ that the `(int32_t)` cast is not necessary here, and that `-(uint32_t)(nTableMask)` gives the same result without being implementation defined.